lite2/client.go

```diff
@@ -649,15 +650,34 @@ func (c *Client) cleanup(stopHeight int64) error {
 		return errors.Wrap(err, "can't get last trusted height")
 	}
 
-	// 3) Remove all headers and validator sets.
-	if stopHeight < oldestHeight {
-		stopHeight = oldestHeight
+	// 3) Ensure the start height and stop height are within the range of headers stored.
+	if stopHeight == 0 || stopHeight > latestHeight {
+		stopHeight = latestHeight
+	}
+	if startHeight < oldestHeight {
+		startHeight = oldestHeight
 	}
-	for height := stopHeight; height <= latestHeight; height++ {
-		err = c.trustedStore.DeleteSignedHeaderAndValidatorSet(height)
+	if startHeight > stopHeight {
+		return errors.Errorf("startHeight of %d must be less than stopHeight %d", startHeight, stopHeight)
+	}
+
+	for startHeight <= stopHeight {
+		// 4a) Delete the header and next validator set
+		err = c.trustedStore.DeleteSignedHeaderAndValidatorSet(startHeight)
 		if err != nil {
-			c.logger.Error("can't remove a trusted header & validator set", "err", err, "height", height)
-			continue
+			c.logger.Error("can't remove a trusted header & validator set", "err", err, "height",
+				startHeight)
+		}
+		// 4b) Jump to the next height
+		if startHeight < stopHeight {
+			h, err := c.trustedStore.SignedHeaderAfter(startHeight)
+			if err != nil {
+				return errors.Wrap(err, "tried to clean up headers")
+			}
+			startHeight = h.Height
+		} else {
```

Comment:
does this mean there will be only one header left == stopHeight?

Reply:
yup